tuf/api/metadata.py

```diff
@@ -343,7 +343,9 @@ def verify_delegate(
             roles = self.signed.delegations.roles
             # Assume role names are unique in delegations.roles: #1426
             # Find first role in roles with matching name (or None if no match)
-            role = next((r for r in roles if r.name == delegated_role), None)
+            role = next(
+                (r for r in roles.values() if r.name == delegated_role), None
+            )
```

Comment:
I would imagine you would now have a slightly easier way of looking up delegated roles by name? :grin:

please update the comment as well

Reply:
Yeaa... you are right. Updated the pr.